sys/include/riotboot/flashwrite.h

```diff
@@ -224,6 +229,30 @@ static inline int riotboot_flashwrite_finish(riotboot_flashwrite_t *state)
                                           RIOTBOOT_FLASHWRITE_SKIPLEN);
 }
 
+/**
+ * @brief   Invalidate a slot header (riotboot version)
+ *
+ * This function invalidates the target slot.
+ *
+ * @note    If this function is called with only one valid slot,
+ *          The invalidation will fail in order to keep one image
+ *          to run after reboot
+ *
+ * @param[in]   slot       Target slot to invalidate
+ *
+ * @returns     0 on success, <0 otherwise
+ */
+int riotboot_flashwrite_invalidate(int slot);
+
+/**
+ * @brief   Revert a firmware update (riotboot version)
+ *
+ * This function clears the slot having the most recent firmware revision.
```

Comment:
I think we need to be a little more accurate here, it does not clear the slot but only invalidate it writing garbage to the header magic number.

Reply:
This also technically does not revert a firmware update, it just invalidates the running image A reboot is needed to "revert". I think `riotboot_flashwrite_invalidate_newest` or something like that would be more accurate.